lib/segment/src/types.rs

```diff
@@ -1292,38 +1298,126 @@ pub struct GeoRadius {
 }
 
 impl GeoRadius {
-    pub fn check_point(&self, lon: f64, lat: f64) -> bool {
+    pub fn check_point(&self, point: &GeoPoint) -> bool {
         let query_center = Point::new(self.center.lon, self.center.lat);
-        query_center.haversine_distance(&Point::new(lon, lat)) < self.radius
+        query_center.haversine_distance(&Point::new(point.lon, point.lat)) < self.radius
     }
 }
 
+#[derive(Deserialize)]
+pub struct GeoPolygonShadow {
+    pub exterior: GeoLineString,
+    pub interiors: Vec<GeoLineString>,
```

Comment:
What about making `interiors` optional?
It seems to me most people would not use it, it would make the API lighter not having to provide an empty list.

WDYT?

Reply:
yes, I think this is a good idea! Changed to Option and also modify integration test to ensure skipping interiors is okay.